### DISCUSSION TITLE
What is credentials.storage file?

### DISCUSSION THREAD

**Discussion by Dark-Mist** [jonnekaunisto/simple-youtube-api#56] [Q&A]
channel.login("client_secret.json", "credentials.storage")
Please someone explain me, what is this file? what does it contains?

**Comment by p5quared** [comment]
From the source:
`def login(
        self,
        client_secret_path: str,
        storage_path: str,
        scope=youtube_api.SCOPES,
    )`
....
`STORAGE = Storage(storage_path)
        credentials = STORAGE.get()`

"credentials.storage" is a path to a credentials file.
It is from the Oauth2Client library. It is used to validate your login info and then the successful object is passed to the 'build' function from GoogleApiClient and returned to self.channel. 

Whenever you have questions about such things, definitely try using the search function through the source code and work through a few steps, sometimes source is a terrible mess but this project is actually pretty readable.
